### PR TITLE
FreeBSDPkg: Initial implemenation

### PR DESCRIPTION
**Pull request by ThreeFx**
Manages packages via `pkg` on FreeBSD installations. Basically a copy of pkg_openbsd which I adapted to FreeBSD.

I tested it with a spare FreeBSD machine, should work as advertised.

I'm not particularly into Python and/or bundlewrap, so code style comments etc. are much appreciated :-)

### CODE REVIEW

**Comment by ThreeFx** [comment]
I give up. The test passes locally (python3.9 on Debian unstable). Anybody have an idea?

File: bundlewrap/items/pkg_freebsd.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+from bundlewrap.exceptions import BundleError
+from bundlewrap.items import Item
+from bundlewrap.utils.text import mark_for_translation as _
+
+def parse_pkg_name(pkgname, line):
+    # Contains the assumption that version may not contain '-', which is covered
+    # according to the FreeBSD docs (Section 5.2.4, "PKGNAMEPREFIX and PKGNAMESUFFIX")
+    installed_package, _, installed_version = line.rpartition('-')
+    assert installed_package != "", _("Unexpected FreeBSD package name: {line}").format(line=line)
```
Comment (trehn) [inline bundlewrap/items/pkg_freebsd.py:11]: You are redefining `_` here, so it's no longer an alias for `mark_for_translation` (gettext function), but a string returned from `str.rpartition()`.

File: bundlewrap/items/pkg_freebsd.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+from shlex import quote
+
+from bundlewrap.exceptions import BundleError
+from bundlewrap.items import Item
+from bundlewrap.utils.text import mark_for_translation as _
+
+def parse_pkg_name(pkgname, line):
+    # Contains the assumption that version may not contain '-', which is covered
+    # according to the FreeBSD docs (Section 5.2.4, "PKGNAMEPREFIX and PKGNAMESUFFIX")
+    installed_package, _, installed_version = line.rpartition('-')
```
Comment (trehn) [inline bundlewrap/items/pkg_freebsd.py:10]:
```suggestion
    installed_package, _sep, installed_version = line.rpartition('-')
```

You are redefining _ here, so it's no longer an alias for mark_for_translation (gettext function), but a string returned from str.rpartition().

File: bundlewrap/items/pkg_freebsd.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+from shlex import quote
+
+from bundlewrap.exceptions import BundleError
+from bundlewrap.items import Item
+from bundlewrap.utils.text import mark_for_translation as _
+
```
Comment (trehn) [inline bundlewrap/items/pkg_freebsd.py:6]:
```suggestion


```

https://pep8.org/#blank-lines

File: tests/unit/pkg_freebsd.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from bundlewrap.items.pkg_freebsd import parse_pkg_name
+from pytest import raises
+
```
Comment (trehn) [inline tests/unit/pkg_freebsd.py:3]:
```suggestion


```

File: tests/unit/pkg_freebsd.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+def test_not_found():
+    found, version = parse_pkg_name("tree", "zsh-5.8")
+    assert found is False
+
```
Comment (trehn) [inline tests/unit/pkg_freebsd.py:7]:
```suggestion


```

File: tests/unit/pkg_freebsd.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+def test_version():
+    found, version = parse_pkg_name("tree", "tree-1.8.0")
+    assert found is True
+    assert version == "1.8.0"
+
```
Comment (trehn) [inline tests/unit/pkg_freebsd.py:12]:
```suggestion


```

File: tests/unit/pkg_freebsd.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+def test_version_with_epoch():
+    found, version = parse_pkg_name("zsh-syntax-highlighting", "zsh-syntax-highlighting-0.7.1,1")
+    assert found is True
+    assert version == "0.7.1,1"
+
```
Comment (trehn) [inline tests/unit/pkg_freebsd.py:17]:
```suggestion


```

File: tests/unit/pkg_freebsd.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+from bundlewrap.items.pkg_freebsd import parse_pkg_name
```
Comment (trehn) [inline tests/unit/pkg_freebsd.py:1]:
```suggestion
from bundlewrap.items.pkg_freebsd import parse_pkg_name

```

**Comment by trehn** [comment]
Can you also add this to the docs? Should just be copy and paste. Relevant files:

```
docs/content/items/pkg_freebsd.md  # create this, copy from openbsd
docs/content/repo/items.py.md  # extend the table
docs/mkdocs.yml  # add to nav
```

File: bundlewrap/items/pkg_freebsd.py (hunk 2 above)
Comment (ThreeFx) [reply bundlewrap/items/pkg_freebsd.py:10]: Oh, that's it! Thanks, I would never have found that :see_no_evil:

**Comment by trehn** [comment]
Thank you!

**Pull request merged**